sql/src/main/java/io/crate/metadata/sys/ClassifiedHistograms.java

```diff
@@ -54,6 +57,33 @@ public Histogram histogram() {
         public Classification classification() {
             return classification;
         }
+
+        long sumOfDurations() {
+            return sumOfDurations;
+        }
+    }
+
+    private static class Metrics {
+
+        private final ConcurrentHistogram histogram;
+        private final AtomicLong sumOfDurations = new AtomicLong();
+
+        /**
+         * @param highestTrackableValue represents the highest value to track
+         * @param numberOfSignificantValueDigits the precision to use
+         */
+        Metrics(final long highestTrackableValue, final int numberOfSignificantValueDigits) {
+            histogram = new ConcurrentHistogram(highestTrackableValue, numberOfSignificantValueDigits);
+        }
+
+        void recordValue(long duration) {
+            histogram.recordValue(Math.min(Math.max(0, duration), HIGHEST_TRACKABLE_VALUE));
+            sumOfDurations.addAndGet(duration);
```

Comment:
Should we use the result of the `min/max` dance to avoid adding negative durations?

Reply:
maybe I'm not understanding how this works but the histogram has the `min` and `max` for all statements of a particular type and not for one statement no? (hence the `max`-`min` is the duration of the longest running statement of a particular type?)

Reply:
The problem is that we calculate the duration based on `System.currentTimeMillis()` and between end and start there might have been a time adjustement, leading to a negative duration. `recordValue` fails if it is negative (which is why we have this `Math.min(Math.max(0, duration))` dance.) I thought I had also added a comment somewhere about this.

Either way. My suggestion was to do a `long safeDuration = Math.min(Math.max(0, duration))` and use it for both `recordValue` and `addAndGet(duration)`.

(Another option would be that we switch over to using a monotonic counter. But the reason we use `currentTimeMillis` is that we require it for the job-logs entries anyway, so it seemed good to avoid some extra work)

Reply:
Gotcha